### DISCUSSION TITLE
Expo Router in Monorepo crashes and "fast refresh" not working

### DISCUSSION THREAD

**Discussion by ShaneKeney** [expo/router#263] [Q&A]
I have the following repo for demo purposes here:  https://github.com/ShaneKeney/expo-router-monorepo-bug .

This was derived and slimmed down from https://github.com/byCedric/expo-monorepo-example.

**Background:**
I have two app project templates in the Monorepo:  `expo-router-template` and `expo-monorepo-example`.

**The issue I am experiencing is two-fold:**

First, "fast refresh" does not work when `expo-router-template` project is present in the repo.  If you delete the Router template project in `apps` directory, the fast refresh will start working again for `expo-monorepo-example`.  As soon as you add back the `expo-router-template`, fast refresh stops working.

Additionally, if you use `app.config.js` instead of `app.json` for the `expo-router-template` then the app will crash in development when you attempt to bring up the Developer Menu via 3 finger tap and/or "Shake to Refresh".

**Comment by toptal-dave** [comment]
Did you find a solution to this?  I tried to follow the documentation and find that the fast refresh using Typescript with Expo Router doesn't work and it could be related.  I am on version 49 of the SDK though...

**Reply by ShaneKeney** [reply]
I did not find an answer.  I ended up using my project outside of a Monorepo for the time being until I can better understand what is causing the issues above and a couple other issues I was facing.

**Reply by toptal-dave** [reply]
OK thanks!  I ended up just going with Ionic because it better supported web and integrated with SST through Vite much better.

**Comment by anup-a** [comment]
Did anyone find a solution to this?

**Reply by marklawlor** [reply]
This repo is for v2 and only maintained for security issues.

If you are having an issue with Expo Router v3, please create and issue here https://github.com/expo/expo
